bcipy/task/base_calibration.py

```diff
@@ -0,0 +1,324 @@
+        self.wait_screen_message_color = parameters['stim_color']
+
+        self.inquiry_generator = self.init_inquiry_generator()
+        self.display = self.init_display()
+
+        self.session = self.init_session()
+        self.write_session_data()
+
+    @property
+    def enable_breaks(self) -> bool:
+        """Whether to allow breaks"""
+        return self.parameters['enable_breaks']
+
+    @property
+    def symbol_set(self) -> List[str]:
+        """Symbols used in the calibration"""
+        return self._symbol_set
+
+    def name(self) -> str:
+        """Task name"""
+        return f"{self.MODE} Calibration Task"
+
+    def wait(self, seconds: Optional[float] = None) -> None:
+        """Pause for a time.
+
+        Parameters
+        ----------
+        - seconds : duration of time to wait; if missing, defaults to the
+        value of the parameter `'task_buffer_length'`
+        """
+        seconds = seconds or self.parameters['task_buffer_length']
+        core.wait(seconds)
+
+    def init_display(self) -> Display:
+        """Initialize the display"""
+        raise NotImplementedError
+
+    def init_inquiry_generator(self) -> Iterator[Inquiry]:
+        """Initializes a generator that returns inquiries to be presented."""
+        parameters = self.parameters
+        schedule = generate_calibration_inquiries(
+            self.symbol_set,
+            inquiry_count=parameters['stim_number'],
+            stim_per_inquiry=parameters['stim_length'],
+            stim_order=StimuliOrder(parameters['stim_order']),
+            jitter=parameters['stim_jitter'],
+            target_positions=TargetPositions(parameters['target_positions']),
+            percentage_without_target=parameters['nontarget_inquiries'],
+            timing=[
+                parameters['time_prompt'], parameters['time_fixation'],
+                parameters['time_flash']
+            ],
+            color=[
+                parameters['target_color'], parameters['fixation_color'],
+                parameters['stim_color']
+            ])
+        return (Inquiry(*inq) for inq in schedule.inquiries())
+
+    def init_session(self) -> session_data.Session:
+        """Initialize the session data."""
+        return session_data.Session(save_location=self.file_save,
+                                    task='Calibration',
+                                    mode=self.MODE,
+                                    symbol_set=self.symbol_set,
+                                    task_data=self.session_task_data())
+
+    def session_task_data(self) -> Optional[Dict[str, Any]]:
+        """"Task-specific session data"""
+        return None
+
+    def trigger_type(self, symbol: str, target: str,
+                     index: int) -> TriggerType:
+        """Trigger Type.
+
+        This method is passed to convert_timing_triggers to properly assign TriggerTypes
+            to the timing of stimuli presented.
```

Comment:
Would be helpful to document the assumptions here. 

`target prompt, fixation, stimuli (nontarget or target; dep on target`